### ISSUE TITLE
`unnecessary_nesting_linter()` incorrectly recommends to get rid of braces in ggplot2 chains

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2852]
## Minimal Reprex

``` r
"library(ggplot2)
x <- TRUE
ggplot(mtcars, aes(wt, mpg)) +
  {
    if (x) geom_point()
  } +
  geom_smooth()" -> code

library(lintr)
lint(text = code, linters = unnecessary_nesting_linter())
#> <text>:4:3: warning: [unnecessary_nesting_linter] Reduce the nesting of this statement by removing the braces {}.
#>   {
#>   ^
```

<sup>Created on 2025-04-23 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

## Code with and without lint

Original code produces expected output

``` r
library(ggplot2)
x <- TRUE
ggplot(mtcars, aes(wt, mpg)) +
  {
    if (x) geom_point()
  } +
  geom_smooth()
#> `geom_smooth()` using method = 'loess' and formula = 'y ~ x'
```

![](https://i.imgur.com/U5Zht0c.png)

Recommended code produces error

``` r
ggplot(mtcars, aes(wt, mpg)) +
  if (x) geom_point() +
  geom_smooth()
#> Error:
#> ! Cannot add <ggproto> objects together.
#> ℹ Did you forget to add this object to a <ggplot> object?
```

<sup>Created on 2025-04-23 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>
